packages/vscode-extension/src/commands/ImportObjects.ts

```diff
@@ -0,0 +1,238 @@
+		const destinationFolder = directoryName.split(this.getProjectFolderPath())[1].replace(/\\/gi, '/');
+		return destinationFolder;
+	}
+
+	private async promptAppId(projectInfoService: { getApplicationId: () => string }) {
+		const filterAppId = await window.showQuickPick(
+			[this.translationService.getMessage(ANSWERS.YES), this.translationService.getMessage(ANSWERS.NO)],
+			{
+				placeHolder: this.translationService.getMessage(IMPORT_OBJECTS.QUESTIONS.FILTER_APP_ID),
+				canPickMany: false,
+			}
+		);
+
+		if (!filterAppId) {
+			return;
+		}
+
+		if (filterAppId && filterAppId == this.translationService.getMessage(ANSWERS.NO)) {
+			return '';
+		}
+
+		const defaultAppId = projectInfoService.getApplicationId();
+		let appId = await window.showInputBox({
+			ignoreFocusOut: true,
+			placeHolder: this.translationService.getMessage(IMPORT_OBJECTS.QUESTIONS.APP_ID, defaultAppId),
+			validateInput: (fieldValue) => {
+				if (fieldValue === '') {
+					return null; //In this case empty field is valid
+				}
+				let validationResult = InteractiveAnswersValidator.showValidationResults(fieldValue, InteractiveAnswersValidator.validateSuiteApp);
+				return typeof validationResult === 'string' ? validationResult : null;
+			},
+		});
+		if (appId == undefined) {
+			return;
+		}
+
+		if (appId.length == 0) {
+			appId = defaultAppId;
+		}
+		return appId;
+	}
+
+	private async promptSelectedScriptId() {
+		let scriptId = await window.showInputBox({
+			ignoreFocusOut: true,
+			placeHolder: this.translationService.getMessage(IMPORT_OBJECTS.QUESTIONS.SCRIPT_ID),
+			validateInput: (fieldValue) => {
+				if (fieldValue === '') {
+					return null; //In this case empty field is valid
+				}
+				let validationResult = InteractiveAnswersValidator.showValidationResults(fieldValue, InteractiveAnswersValidator.validateScriptId);
+				return typeof validationResult === 'string' ? validationResult : null;
+			},
+		});
+
+		return scriptId;
+	}
+
+	private async promptObjectTypes(): Promise<string[] | undefined> {
+		const selectedObjectTypes = await window.showQuickPick(
+			objectTypes.map((objectType) => objectType.value.type),
+			{
+				placeHolder: this.translationService.getMessage(IMPORT_OBJECTS.QUESTIONS.SELECT_TYPES),
+				canPickMany: true,
+			}
+		);
+		return selectedObjectTypes;
+	}
+
+	private async listObjects(appId: string | undefined, selectedObjectTypes: string[], scriptId: string | undefined) {
+		const actionResult = await this.customObjectService.listObjects(appId, selectedObjectTypes, scriptId, this.executionPath);
+		return actionResult;
+	}
+
+	private async promptObjects(objectList: any[]): Promise<string[] | undefined> {
+		const selectedObjects = await window.showQuickPick(
+			objectList.map((object) => object.scriptId),
+			{
+				placeHolder: this.translationService.getMessage(IMPORT_OBJECTS.QUESTIONS.SELECT_OBJECTS),
+				canPickMany: true,
+			}
+		);
+		return selectedObjects;
+	}
```

Comment:
I think we should add a validation that does not allow to continue without selecting at least one object.

Reply:
Quickpick doesn't allow validation, only Inputboxes.
That's why there is the validation in the other chunk of code.

Reply:
We could be asking again and again while the user is not selecting at least one object, that could be a way to force the user to select at least one, or hit ESCAPE to not continue with the process.